### PR TITLE
GH-2373: As a developer I want to replace n4js-runtime-node by @types/node

### PR DESCRIPTION
**Pull request by mor-n4**
See #2373.

### CODE REVIEW

File: plugins/org.eclipse.n4js.model/src/org/eclipse/n4js/AnnotationDefinition.java [hunk 1]
```diff
@@ -258,6 +259,29 @@ public final class AnnotationDefinition {
...
+			.targets(SCRIPT)
+			.args(STRING_LITERAL)
+			.transitive()
+			.retention(RetentionPolicy.TYPE)
+			.end();
+
+	/**
+	 * States that the annotated {@link TClassifier classifier} owns an index signature on the .d.ts side.
+	 * <p>
+	 * Only to be used in .d.ts.
```
Comment (mmews-n4) [inline plugins/org.eclipse.n4js.model/src/org/eclipse/n4js/AnnotationDefinition.java:277]: Do we need a validation for that constraint? I.e. unsupported feature validation?

**Review by mmews-n4 — APPROVED** [review]
👍  Really great improvements!

File: plugins/org.eclipse.n4js.model/src/org/eclipse/n4js/AnnotationDefinition.java (hunk 1 above)
Comment (mor-n4) [reply plugins/org.eclipse.n4js.model/src/org/eclipse/n4js/AnnotationDefinition.java:277]: Good idea, will do!

**Comment by mor-n4** [comment]
Thanks for the review @mmews-n4!

Build is green, merging ...

**Pull request merged**